frontend/packages/component-library/src/tags/_Tag.tsx

```diff
@@ -32,39 +33,81 @@ export interface TagProps {
    *  Icon object consists of icon(icon name/style, title for screenReader,
    *  and the placement of the icon (left or right))*/
   icon?: TagIconProps;
+  /** Unique key */
+  key?: React.Key;
 }
 
-const Tag: React.FunctionComponent<TagProps> = ({
-  label,
-  tooltipContent,
-  tooltipId,
-  icon,
-}) => {
-  return tooltipContent && tooltipId ? (
-    <WithTooltip
-      tooltipProps={{
-        direction: 'onTop',
-        text: tooltipContent,
-        tooltipId,
-      }}
-    >
-      <div
-        tabIndex={0} // eslint-disable-line jsx-a11y/no-noninteractive-tabindex
-        aria-describedby={tooltipId}
-        className={moduleStyles.tag}
-      >
+export interface DefaultTagProps extends BaseTagProps {
+  type?: 'default';
+}
+
+export interface CloseTagProps extends BaseTagProps {
+  type: 'closable';
+  /** onClose callback gives the tag an accessible close button on the
+   * right side of the label */
+  onClose: () => void;
+}
+
+export type TagProps = DefaultTagProps | CloseTagProps;
+
+const Tag: React.FunctionComponent<TagProps> = props => {
+  const {
+    label,
+    ariaLabel,
+    tooltipContent,
+    tooltipId,
+    icon,
+    type = 'default',
+  } = props;
+  const tooltipWrapper = useCallback(
+    (children: React.ReactNode) =>
+      tooltipContent && tooltipId ? (
+        <WithTooltip
+          tooltipProps={{
+            direction: 'onTop',
+            text: tooltipContent,
+            tooltipId: tooltipId,
+          }}
+        >
+          {children}
+        </WithTooltip>
+      ) : (
+        children
+      ),
+    [tooltipContent, tooltipId],
+  );
+
+  const containerAttrs: HTMLAttributes<HTMLDivElement> = {
+    className: moduleStyles.tag,
+    tabIndex: 0,
+    'aria-label': ariaLabel,
+    'aria-describedby': tooltipContent && tooltipId ? tooltipId : undefined,
+  };
+
+  if (type === 'closable') {
+    delete containerAttrs.tabIndex;
+    const {onClose} = props as CloseTagProps;
+    return tooltipWrapper(
+      <div {...containerAttrs}>
+        <span>{label}</span>
```

Comment:
(not blocking, can be done as a follow up pr/discussion)
How about making it like this?
```suggestion
        {/* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */}
        <span tabIndex={0}>{label}</span>
```
I know it's not ideal, but it will make tooltip be keyboard and screen reader accessible and we won't be nesting focusable elements (cause with current solution only close button will be accessible)?

Also, as another option - maybe we can remove tooltips usage for closable tags?

Here's a video of how it will look like:

https://github.com/user-attachments/assets/dc64d778-9448-4e16-8922-5f3548a70f43



cc @moshebaricdo @hannahbergam @markabarrett

Reply:
yeah, I really wasn't sure what we should do here. I felt like having to tab twice to get through an element was bad, and the tooltip does still pop when the close button gets focus 🤷

Reply:
Hmm, agreed that this isn't super ideal, it's odd to see the tooltip appear both times, is it possible to have the tooltip appear when content/label is focused, but not for the close button? I don't want to get rid of tooltip support entirely for this tag type, maybe it becomes optional so we can decide depending on context.

Reply:
here's a video of the existing implementation to compare:


https://github.com/user-attachments/assets/fe9106d5-406d-4b6b-b5a9-985f3208bd7e

Reply:
Ok I dug through about 9 systems that all had a removable tag type, and it seems like they all take the approach of only making the close button focusable and not the label—and they all do not use tooltips for this case lol.

Reply:
the tooltip is definitely optional (only shows if you pass the correct props, tooltipContent and tooltipId)

as far as preventing it from showing when the close button has focus, I'd have to dig into the tooltip implementation, but I'm pretty sure it would be tricky. the tooltip is a wrapper for any component